### PR TITLE
Minor post-platforms fix.

### PR DESCRIPTION
**Pull request by hjoliver**
This is a small change with no associated Issue.

Found some problems while looking at #3706 

- `TaskEventsManager` has a method `get_host_conf` that used to extract config items from the deprecated task `remote` and `job` sections, defaulting to global config.  It's broken because **some** deprecated `job` items get auto-upgraded to top level task settings (so they no longer have a secondary dict key), but `remote` items don't because they've moved to platforms (so in deprecated form they still have a secondary dict key). So I've split these into two separate functions.
- `TaskJobManager` also makes a call to the above event manager method to extract `batch systems` config (`err tailer` etc.) which is now in platforms. And, the `batch_sys_config` dict is not actually used anywhere, so I've removed it.

@wxtim - could you cast your "platforms eye" over this to see I'm on the right track?

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Comment by hjoliver** [comment]
(Draft until I think about tests)

**Comment by wxtim** [comment]
It looks reasonable - Sorry that I missed doing this.
these look like nicely testable units, but I guess that you probably want to check that over-all functionality hasn't changed in an unexpected way so you want a functional test?

File: cylc/flow/task_events_mgr.py [hunk 1]
```diff
@@ -252,18 +252,35 @@ def check_job_time(self, itask, now):
...
-            ret = itask.tdef.rtconfig[skey][key]
+        SKEY = 'remote'
+        if SKEY in overrides and overrides[SKEY].get(key) is not None:
+            ret = overrides[SKEY][key]
+        elif itask.tdef.rtconfig[SKEY].get(key) not in (None, []):
+            ret = itask.tdef.rtconfig[SKEY][key]
         else:
-            try:
-                ret = itask.platform[key]
-            except (KeyError, ItemNotFoundError):
```
Comment (hjoliver) [inline cylc/flow/task_events_mgr.py:265]: No way to get `ItemNotFoundError` here

File: cylc/flow/task_events_mgr.py [hunk 2]
```diff
@@ -252,19 +252,27 @@ def check_job_time(self, itask, now):
...
+        """Get deprecated "[remote]" items that default to platforms."""
         overrides = self.broadcast_mgr.get_broadcast(itask.identity)
-        if skey in overrides and overrides[skey].get(key) is not None:
-            ret = overrides[skey][key]
-        elif itask.tdef.rtconfig[skey].get(key) not in (None, []):
-            ret = itask.tdef.rtconfig[skey][key]
-        else:
-            try:
-                ret = itask.platform[key]
-            except (KeyError, ItemNotFoundError):
```
Comment (hjoliver) [inline cylc/flow/task_events_mgr.py:265]: There's no way to get `ItemNotFoundError` here.  And `KeyError` can only occur if an illegal config item is requested.

File: tests/unit/test_task_events_mgr_2.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+from unittest.mock import Mock
+import pytest
+
+
+from cylc.flow.task_events_mgr import TaskEventsManager
+
+
+@pytest.mark.parametrize(
+    "broadcast, remote, platforms, expected",
```
Comment (kinow) [inline tests/unit/test_task_events_mgr_2.py:25]: Nothing wrong here, but my OCD makes me keeping looking at the parameter names are out of order here and in the function declaration🤣

**Review by kinow — APPROVED** [review]
Tests look good to me! LGTM

File: tests/unit/test_task_events_mgr_2.py (hunk 3 above)
Comment (hjoliver) [reply tests/unit/test_task_events_mgr_2.py:25]: Good spotting. I had reordered the parameters at one point but forgot to do the same for the function args. It doesn't seem to matter, but I fixed it anyway.

**Comment by kinow** [comment]
One functional test job failing. Strange. Could it be similar issue to that from my PR fixing unused imports??

**Comment by hjoliver** [comment]
Damn, it passed before my final trivial commit.

**Comment by hjoliver** [comment]
It is a flaky one. I've restarted the tests.

**Comment by kinow** [comment]
> It is a flaky one. I've restarted the tests.

Two jobs failing now :smile:

**Review by wxtim — APPROVED** [review]
Oh dear - tests look flaky this morning

**Comment by hjoliver** [comment]
The test failure here appears to be genuine, but is also happening on master (local testing): 
```
tests/f/cylc-poll/04-poll-multi-hosts.t (Wstat: 0 Tests: 3 Failed: 2)
  Failed tests:  2-3
```
So I'll merge this and try to diagnose the problem on master.

**Pull request merged**